src/utils/tree.js

```diff
@@ -0,0 +1,28 @@
+export const buildTree = (arr, targetId, depth) => {
+  if (arr.length) {
+    return `<ul data-depth=${depth}>${arr
+      .map(
+        ({ id, title, documents }) =>
+          `<li>${
+            documents.length
+              ? `<label><ion-icon name="chevron-down-outline" class="arrow"/></label>`
+              : "- "
+          }<span class="documentTitle ${
+            Number(targetId) === id ? "selected" : ""
+          }" data-id="${id}">${title}</span><button data-add="${id}"> + </button><button data-remove="${id}"> - </button>
+          </li>${buildTree(documents, targetId, depth + 1)}`
+      )
+      .join("")}</ul>`;
+  } else return `<ul data-depth=${depth}></ul>`;
```

Comment:
배열의 길이를 딱히 검사하지 않아도 될 것 같습니다. 검사하지 않아도 똑같이 작동됩니다. 그리고, `arr` 같은 줄임말에 의미가 모호한 단어 대신 `documents` 같은 단어도 좋을 것 같습니다. 또, buildTree라는 함수가 굉장히 보편적으로 사용할 수 있을 것 같은 이름인데, 실제로는 문서들의 정보를 담은 돔 요소를 문자열로 조립하여 반환하고 있습니다. `createDocumentHTML` 같은 이름도 좋을 것 같고, 해당 함수가 utils에 있어야할 이유도 잘 모르겠습니다. utils에 들어있는 함수들은 다른 여러 컴포넌트에서 사용할 수 있어야 할 것 같은데, 현재로서는 SideMenu 안에서만 사용될 것 같아요.
```suggestion
  return `<ul data-depth=${depth}>${arr
    .map(
      ({ id, title, documents }) => 
        `<li>${
          documents.length
            ? `<label><ion-icon name="chevron-down-outline" class="arrow"/></label>`
            : "- "
        }<span class="documentTitle ${
          Number(targetId) === id ? "selected" : ""
        }" data-id="${id}">${title}</span><button data-add="${id}"> + </button><button data-remove="${id}"> - </button>
        </li>${buildTree(documents, targetId, depth + 1)}`
    )
    .join("")}</ul>`;
```